### PR TITLE
Removed trailing zero in WASI::fd_prestat_dir_name name return (for #3025)

### PR DESCRIPTION
**Pull request by ptitSeb**
# Description
Removed trailing zero in WASI::fd_prestat_dir_name returned name (for #3025)

### CODE REVIEW

**Comment by epilys** [comment]
[Posix defines character strings as null terminated](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap03.html#tag_03_276), personally I'd like all path strings that go out into a module to be null terminated. What do you think?

**Comment by ptitSeb** [comment]
I agree, but this is WASI here, not POSIX. check the ticket mentioned for more detail. Not sure the change should be done. But if we don't it, we should close the mentioned ticket as a "will not be done"

File: lib/wasi/src/syscalls/mod.rs [hunk 1]
```diff
@@ -959,13 +959,11 @@ pub fn fd_prestat_dir_name<M: MemorySize>(
...
     match guard.deref() {
         Kind::Dir { .. } | Kind::Root { .. } => {
-            // TODO: verify this: null termination, etc
             let path_len: u64 = path_len.into();
-            if (inode_val.name.len() as u64) < path_len {
+            if (inode_val.name.len() as u64) <= path_len {
                 wasi_try_mem!(path_chars
                     .subslice(0..inode_val.name.len() as u64)
                     .write_slice(inode_val.name.as_bytes()));
-                wasi_try_mem!(path_chars.index(inode_val.name.len() as u64).write(0));
```
Comment (syrusakbary) [inline lib/wasi/src/syscalls/mod.rs:968]: We might still want to terminate it by 0 (but just not pass it to the user).
Basically, all the PR as is but, reverting this to what it was:
```
                wasi_try_mem!(path_chars.index(inode_val.name.len() as u64).write(0));
```

Thoughts?

Comment (ptitSeb) [reply lib/wasi/src/syscalls/mod.rs:968]: we might overwite memory here! The memory is allocated client side, for the result of `prestat_fd`, where the 0 is not counted anymore on this PR.
We cannot have both. It with or without the appended 0

Comment (syrusakbary) [reply lib/wasi/src/syscalls/mod.rs:968]: Ok!

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**